vms/platformvm/block/executor/manager.go

```diff
@@ -39,6 +40,10 @@ type Manager interface {
 	// VerifyTx verifies that the transaction can be issued based on the currently
 	// preferred state. This should *not* be used to verify transactions in a block.
 	VerifyTx(tx *txs.Tx) error
+
+	// VerifyUniqueInputs verifies that the inputs are not duplicated in the
+	// provided blk or any of its ancestors pinned in memory.
+	VerifyUniqueInputs(blkID ids.ID, inputs set.Set[ids.ID]) error
```

Comment:
Was going to ask about this in the previous PR I reviewed but figured it was in flight.